anndata/tests/test_io_utils.py

```diff
@@ -46,7 +46,7 @@ def test_write_error_info(diskfmt, tmp_path):
     # Assuming we don't define a writer for tuples
     a = ad.AnnData(uns={"a": {"b": {"c": (1, 2, 3)}}})
 
-    with pytest.raises(Exception, match=r"Above error raised while writing key 'c'"):
+    with pytest.raises(Exception, match=r"Above error raised while writing key 'b'"):
```

Comment:
I think is correct for what happens now actually - https://github.com/scverse/anndata/pull/851/files#diff-bf0c45f88b00900e7fb8a82d25fd170f8a39c195579ad2b516643a25d81770caR216 has the wrapper applied after returning the `partial` but the error is occuring before that function is ever returned here: https://github.com/scverse/anndata/pull/851/files#diff-bf0c45f88b00900e7fb8a82d25fd170f8a39c195579ad2b516643a25d81770caR208 so it makes sense the error is "one level higher"